app/models/physical_storage.rb

```diff
@@ -46,6 +47,47 @@ def refresh_ems
     EmsRefresh.queue_refresh(ext_management_system)
   end
 
+  def raw_delete_physical_storage
+    raise NotImplementedError, _("raw_delete_physical_storage must be implemented in a subclass")
+  end
+
+  def ensure_delete_valid
+    unless ext_management_system
+      raise _("No Provider defined")
+    end
+    unless ExtManagementSystem.find_by(:id => ems_id)
+      raise ArgumentError, _("ext_management_system cannot be found")
+    end
+    unless "#{ext_management_system.type.to_s}::PhysicalStorage".constantize.supports?("delete")
+      raise _("#{ext_management_system.name} provider does not supports physical storage deletion")
+    end
+  end
+
+  def delete_physical_storage
+    raw_delete_physical_storage
+  end
+
+  # Delete a storage system as a queued task and return the task id. The queue
+  # name and the queue zone are derived from the EMS, and a userid is mandatory.
+  def delete_physical_storage_queue(userid)
+    ensure_delete_valid
+
+    task_opts = {
+        :action => "deleting PhysicalStorage for user #{userid}",
```

Comment:
Hard to tell in github but this looks like 4 spaces instead of 2

Reply:
fixed